routing/record/validation.go

```diff
@@ -100,3 +112,18 @@ func CheckRecordSig(r *pb.Record, pk ci.PubKey) error {
 	}
 	return nil
 }
+
+type RecordSorter []*pb.Record
+
+func (rs RecordSorter) Less(i, j int) bool {
+	mv, _ := MoreValid(rs[j], rs[i])
```

Comment:
how do you reconcile those two? a time on the blockchain in practice correlates with NTP time but is not guaranteed to. a longer blockchain could "show up" at any moment affecting all "expected time per block" calculations. things "may work" for now but will have really bad failure cases when peoples different validity schemes start conflicting (like the commit-chain vs "latest timestamp").